perllib/FixMyStreet/App/Controller/Open311/Updates.pm

```diff
@@ -0,0 +1,87 @@
+package FixMyStreet::App::Controller::Open311::Updates;
+
+use utf8;
+use Moose;
+use namespace::autoclean;
+use Open311;
+use Open311::GetServiceRequestUpdates;
+
+BEGIN { extends 'Catalyst::Controller'; }
+
+=head1 NAME
+
+FixMyStreet::App::Controller::Open311::Updates - Catalyst Controller
+
+=head1 DESCRIPTION
+
+=head1 METHODS
+
+=cut
+
+sub receive : Regex('^open311/v2/servicerequestupdates.(xml|json)$') : Args(0) {
+    my ( $self, $c ) = @_;
+    $c->stash->{format} = $c->req->captures->[0];
+
+    $c->detach('bad_request', [ 'POST' ]) unless $c->req->method eq 'POST';
+
+    my $body;
+    if ($c->cobrand->can('council_area_id')) {
+        $body = $c->cobrand->body;
+    } else {
+        $body = $c->model('DB::Body')->find({ id => $c->get_param('jurisdiction_id') });
+    }
+    $c->detach('bad_request', ['jurisdiction_id']) unless $body;
+    my $user = $body->comment_user;
+
+    my $key = $c->get_param('api_key') || '';
+    my $token = $c->cobrand->feature('open311_token') || '';
+    $c->detach('bad_request', [ 'api_key' ]) unless $key && $key eq $token;
+
+    my $request = {
+        media_url => $c->get_param('media_url'),
+    };
+    foreach (qw(service_request_id update_id updated_datetime status description)) {
```

Comment:
Would be good to reject invalid `status` values. We should support `external_status_code` too, in case there are response templates which are triggered by that.

Reply:
We don't reject invalid statuses we pull, so is at least consistent :) I thought it'd be better to accept invalid values in order to get the update, as I think we handle unknown states quite well.
Have added external_status_code.